tests/components/motionblinds_ble/test_config_flow.py

```diff
@@ -0,0 +1,165 @@
+        result["flow_id"],
+        {const.CONF_MAC_CODE: TEST_MAC},
+    )
+
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["step_id"] == "confirm"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {const.CONF_BLIND_TYPE: const.MotionBlindType.ROLLER},
+    )
+
+    assert result["type"] is data_entry_flow.FlowResultType.CREATE_ENTRY
+    assert result["title"] == f"MotionBlind {TEST_MAC.upper()}"
+    assert result["data"] == {
+        const.CONF_ADDRESS: TEST_ADDRESS,
+        const.CONF_LOCAL_NAME: TEST_NAME,
+        const.CONF_MAC_CODE: TEST_MAC.upper(),
+        const.CONF_BLIND_TYPE: TEST_BLIND_TYPE,
+    }
+    assert result["options"] == {}
+
+    # Second setup to test already configured error
+    result = await hass.config_entries.flow.async_init(
+        const.DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
+
+
+async def test_config_flow_manual_errors(
+    hass: HomeAssistant, motionblinds_ble_connect
+) -> None:
+    """Errors during flow manually initialized by the user."""
+    result = await hass.config_entries.flow.async_init(
+        const.DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {const.CONF_MAC_CODE: "ab:cd:ef:gh"},
+    )
+
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {"base": const.ERROR_INVALID_MAC_CODE}
+
+    with patch(
+        "homeassistant.components.motionblinds_ble.config_flow.bluetooth.async_scanner_count",
+        return_value=0,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {const.CONF_MAC_CODE: TEST_MAC},
+        )
+
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {"base": const.ERROR_NO_BLUETOOTH_ADAPTER}
+
+    motionblinds_ble_connect[1].name = "WRONG_NAME"
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {const.CONF_MAC_CODE: TEST_MAC},
+    )
+
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {"base": const.ERROR_COULD_NOT_FIND_MOTOR}
+
+    motionblinds_ble_connect[0].discover.return_value = []
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {const.CONF_MAC_CODE: TEST_MAC},
+    )
+
+    assert result["type"] == data_entry_flow.FlowResultType.FORM
```

Comment:
might be `ABORT` here, or how could you recover if you didn't find devices

Reply:
yes (be sure to set a reason though). Yes, also in case of the bluetooth adapters, basically always when you've reached a terminal state, because you can't recover by changing a setting

Reply:
Thanks fixed it, it now aborts instead of showing an error message for NoBluetoothAdapter and NoDevicesFound